packages/next/src/lib/metadata/default-metadata.tsx

```diff
@@ -1,12 +1,17 @@
 import React from 'react'
 import type { ResolvedMetadata } from './types/metadata-interface'
 
-export const createDefaultMetadata = (): ResolvedMetadata => {
+export function createDefaultMetadata(): ResolvedMetadata {
+  let defaultMetadataBase = new URL('http://n')
+  if (process.env.NODE_ENV === 'production' && process.env.VERCEL_URL) {
+    defaultMetadataBase = new URL(`https://${process.env.VERCEL_URL}`)
+  }
```

Comment:
```suggestion
  let defaultMetadataBase
  if (process.env.NODE_ENV === 'production' && process.env.VERCEL_URL) {
    defaultMetadataBase = new URL(`https://${process.env.VERCEL_URL}`)
  } else {
    defaultMetadataBase = new URL('http://n')
  }
```

Reply:
Also - does this mean that if you are self-hosting, the default will be `new URL('http://n')` and then fallbacks to `localhost:3000`? I think we should error in that case if it's under production.

Reply:
If we have to make a decision between the two, IMO "inconsistent and error in build" is better than "consistent and passed build but wrong". And usually large projects will not be build on local machines.